src/distilabel/steps/task/self_instruct.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from typing import TYPE_CHECKING, Any, Dict, List, Union
+
+from jinja2 import Template
+from pydantic import PrivateAttr
+
+from distilabel.steps.task.base import Task
+
+if TYPE_CHECKING:
+    from distilabel.steps.task.typing import ChatType
+
+_SELF_INSTRUCT_TEMPLATE = """
```

Comment:
Could you move this to an external `*.jinja2` template file? Feel free to use `UltraFeedback` as a reference from the `core-refactor` branch